### PR TITLE
yum: is_installed doesn't require repoq anymore

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY
`is_installed` does not use repoquery anymore so it doesn't make sense to decide whether to use yum api or the rpm command according to the repoquery presence. Since both yum api and rpm are required to run the module, run preferably yum api (since it's a bit faster). The rpm command is used only when listing installed packages since it provides a similar output to repoquery which is used by other methods used for listing packages.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
yum

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by gundalow** [comment]
```
                          Traceback (most recent call last):
  File "/tmp/ansible_nJoTGC/ansible_module_yum.py", line 1344, in <module>
    main()
  File "/tmp/ansible_nJoTGC/ansible_module_yum.py", line 1335, in main
    skip_broken, update_only, security, params['installroot'], allow_downgrade)
  File "/tmp/ansible_nJoTGC/ansible_module_yum.py", line 1231, in ensure
    res = remove(module, pkgs, repoq, yum_basecmd, conf_file, en_repos, dis_repos, installroot=installroot)
  File "/tmp/ansible_nJoTGC/ansible_module_yum.py", line 876, in remove
    installed = is_installed(module, pkg, conf_file, en_repos=en_repos, dis_repos=dis_repos, installroot=installroot)
TypeError: is_installed() takes at least 4 arguments (6 given)
```

**Comment by mkrizek** [comment]
cc @resmo

File: lib/ansible/modules/packaging/os/yum.py [hunk 1]
```diff
@@ -248,7 +248,6 @@
 BUFSIZE = 65536
 
 def_qf = "%{epoch}:%{name}-%{version}-%{release}.%{arch}"
-rpmbin = None
```
Comment (abadger) [inline lib/ansible/modules/packaging/os/yum.py:251]: We should leave this.  The global rpmbin is basically a cache so we don't have to pay the cost of looking for the rpm command inside of loops.  (the install() function calls is_installed() inside of a loop, for instance).

File: lib/ansible/modules/packaging/os/yum.py [hunk 2]
```diff
@@ -343,15 +342,15 @@ def is_group_env_installed(name):
     return False
 
 
-def is_installed(module, repoq, pkgspec, conf_file, qf=None, en_repos=None, dis_repos=None, is_pkg=False, installroot='/'):
+def is_installed(module, pkgspec, conf_file, rpm=False, qf=None, en_repos=None, dis_repos=None, is_pkg=False, installroot='/'):
```
Comment (abadger) [inline lib/ansible/modules/packaging/os/yum.py:345]: I don't think it's safe to change when we use the yum api, however I don't remember the specifics.  I would not change this behaviour just to change this as I think it exists for some cornercases.

**Comment by abadger** [comment]
needs_revision

If you want to set the value of the new rpm parameter by checking whether repoq is set prior to caling the is_installed() function, I think that would be an okay cleanup.  Changing that, though, should not be done unless it's fixing a definite bug as there's been bugs in the past which we've solved by choosing one or the other of these code paths in certain cornercases.

**Comment by mkrizek** [comment]
@abadger okay, this was not to fix a bug but a result of me going through the code and thinking this could be changed but as per:

> I don't think it's safe to change when we use the yum api, however I don't remember the specifics. I would not change this behaviour just to change this as I think it exists for some cornercases.

I am going to close it. Thanks for taking a look at this!

**Pull request closed without merging**